config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/Encass.java

```diff
@@ -156,4 +160,34 @@ public void sortArgumentsAndResults() {
         setArguments(getArguments().stream().collect(toCollection(() -> new TreeSet<>(Comparator.comparing(EncassArgument::getName)))));
         setResults(getResults().stream().collect(toCollection(() -> new TreeSet<>(Comparator.comparing(EncassResult::getName)))));
     }
+
+    @Override
+    public AnnotatedEntity<GatewayEntity> getAnnotatedEntity(final String projectName,
+                                                             final String projectVersion) {
+        return annotatedEntity == null ? createAnnotatedEntity(annotations, projectName, projectVersion) :
+                annotatedEntity;
+    }
+
+    public void populateBundleInfo(final AnnotatedEntity<GatewayEntity> annotatedEntity, final Annotation bundleAnnotation, final String projectName,
```

Comment:
I suggest you to move this logic into AnnotatedEntity.
Once you get the annotated entity, you will call some method where you inject additional details like project name, version, etc.